codechecker_common/checker_labels.py

```diff
@@ -0,0 +1,262 @@
+                            '{checker}.')
+
+    def __get_analyzer_data(self, analyzers: Iterable[str]):
+        for analyzer, checkers in self.__data.items():
+            if analyzers is None or analyzer in analyzers:
+                yield analyzer, checkers
+
+    def checkers_by_labels(
+            self,
+            filter_labels: Iterable[str],
+            analyzers: Iterable[str] = None) -> List[str]:
+        """
+        Returns a list of checkers that have at least one of the specified
+        labels.
+
+        filter_labels -- A string list which contains labels with specified
+                         values. E.g. ['profile:default', 'severity:high'].
+        analyzers -- A list of analyzers among which checkers are searched.
+                     By default all analyzers are searched.
+        """
+        collection = []
+
+        filter_labels = set(map(self.__get_label_key_value, filter_labels))
+
+        for _, checkers in self.__get_analyzer_data(analyzers):
+            for checker, labels in checkers.items():
+                labels = set(map(self.__get_label_key_value, labels))
+
+                if labels.intersection(filter_labels):
+                    collection.append(checker)
+
+        return collection
+
+    def label_of_checker(
+            self,
+            checker: str,
+            label: str,
+            analyzers: Iterable[str] = None) -> Union[str, Iterable[str]]:
+        """
+        If a label has unique constraint then this function retuns the value
+        that belongs to the given label or the default value that is set among
+        label constraints. If there is no unique constraint then an iterable
+        object returns with the values assigned to the given label. If the
+        checker name is not found in the label config file then its prefixes
+        are also searched. For example "clang-diagnostic" in the config file
+        matches "clang-diagnostic-unused-argument".
+        """
+        labels = chain.from_iterable([])
+
+        for analyzer, _ in self.labels_of_checker(checker, analyzers):
+            if analyzers is None or analyzer in analyzers:
+                labels = chain(labels, filter(
+                    lambda lab: lab[0] == label,
+                    self.labels_of_checker(checker)))
+
+        if label in CheckerLabels.UNIQUE_LABELS:
+            try:
+                return next(labels)[1]
+            except Exception:
+                return CheckerLabels.UNIQUE_LABELS[label]
+
+        return map(lambda _, value: value, labels)
+
+    def severity(self, checker: str, analyzers: Iterable[str] = None) -> str:
+        """
+        Shorthand for the following call:
+        checker_labels.label_of_checker(checker, 'severity', analyzers)
+        """
+        return self.label_of_checker(checker, 'severity', analyzers)
+
+    def labels_of_checker(
+            self,
+            checker: str,
+            analyzers: Iterable[str] = None) -> List[str]:
```

Comment:
What happens if no analyzer name is specified but the given checker name can be found in multiple label files. Can you please add a test for this use case too?

Reply:
I missed your question here, sorry. In that case the checkers of all analyzers are taken into account. I'll write a test on it too soon.